packages/sanity/src/router/RouteScope.tsx

```diff
@@ -97,13 +103,31 @@ export const RouteScope = function RouteScope(props: RouteScopeProps): React.JSX
     [parent_resolvePathFromState, resolveNextParentState],
   )
 
-  const navigate = useCallback(
-    (nextState: RouterState | null, options?: NavigateOptions) =>
-      parent_navigate(
-        // options?.stickyParams && nextState === null ? null :
-        resolveNextParentState(nextState),
-        options,
-      ),
+  const navigate: RouterContextValue['navigate'] = useCallback(
+    (nextStateOrOptions: NextStateOrOptions, maybeOptions?: NavigateOptions) => {
+      // Check if it's the options-only pattern
+      if (isNavigateOptions(nextStateOrOptions) && !maybeOptions) {
```

Comment:
as an alternative, it should be safe to check the arity of the function as well, and treat it as `options` if it's passed 1 arg and `state` if 2. In theory, I guess we could run into trouble now if you do `navigate({state: 'foo'})`